src/DatePicker/DatePicker.js

```diff
@@ -206,13 +209,55 @@ class DatePicker extends Component {
         date: date,
       });
     }
+
+    setTimeout(() => {
+      this.setState({
+        isFocused: false,
+      });
+    }, 0);
+    this.refs.input.focus();
+
     if (this.props.onChange) {
       this.props.onChange(null, date);
     }
   };
 
+  handleBlur = (event, shouldSetFocus) => {
+    if (shouldSetFocus) {
+      this.refs.input.focus();
+    }
+
+    if (this.state.isFocused) {
+      this.setState({
+        isFocused: false,
+      }, this.refs.dialogWindow.dismiss);
+    }
+
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  };
+
+  handleDismiss = () => {
+    if (this.state.isFocused) {
+      this.refs.input.focus();
+      this.setState({
+        isFocused: false,
+      });
+    }
+
+    if (this.props.onDismiss) {
```

Comment:
was thinking...this and the `onFocus` methods, do you think they more rightly belong inside the if statements